plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -935,9 +941,83 @@ public Optional<ConnectorTableHandle> applyDelete(ConnectorSession session, Conn
     }
 
     @Override
-    public ConnectorTableHandle beginDelete(ConnectorSession session, ConnectorTableHandle tableHandle)
+    public ConnectorTableHandle beginDelete(ConnectorSession session, ConnectorTableHandle tableHandle, RetryMode retryMode)
+    {
+        IcebergTableHandle table = (IcebergTableHandle) tableHandle;
+        if (table.getFormatVersion() < 2) {
+            throw new TrinoException(NOT_SUPPORTED, "Iceberg table updates require at least format version 2");
+        }
+        String deleteMode = table.getStorageProperties().get(DELETE_MODE);
+        if (!"merge-on-read".equals(deleteMode)) {
+            String errorMessage = "Row level deletes are only supported for tables with '" + DELETE_MODE + "' set to 'merge-on-read'.";
+            if (deleteMode != null) {
+                errorMessage += " '" + deleteMode + "' is not supported.";
+            }
+            throw new TrinoException(NOT_SUPPORTED, errorMessage);
```

Comment:
if `deleteMode==null` (no specific mode request), it seems correct to carry out the delete operation.
-- we're fulfilling the v2 spec, and not violating any explicit request.
Let's just do this change.

furthermore, when delete mode is set !- merge-on-read. Can this be for performance reasons?
If so, that would be "a soft preference", perhaps set for Spark's benefit (which supports multiple delete modes).  In such case ignoring it and carrying out the delete would be OKish too (best option currently possible).

Can this be set for compatibility reasons? I don't know.

cc @electrum @phd3 @rdblue

Reply:
Actually, i came to a conclusion that this should be viewed as a _tuning property_ only, i.e. advisory. We don't support two modes, so there is nothing to tune, we just just proceed.

Reply:
I agree that this should be advisory. This is a preferred mode.

If a user were to need to enforce this, I think it would only be for the eager rewrite / copy-on-write mode. But that can be enforced by using a v1 table instead of a v2.